force-app/main/default/classes/DMLOperations.cls

```diff
@@ -146,9 +169,46 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        List <Account> AccountList = [SELECT Id, Name FROM Account WHERE Name = :accountName];
+        Account acc;
+        if (AccountList.size() > 0) {
+            acc = AccountList[0];
+        } else {
+            acc = new Account (
+                Name = accountName
+            );
+            insert acc;
+        }
+
+        //Retrieve existing Opportunities
+        List <Opportunity> retrieveOppList = new List <Opportunity> ();
+        retrieveOppList = [SELECT Id, Name FROM Opportunity WHERE AccountId = :acc.Id AND Name IN :oppNames];
```

Comment:
Can you query both the account and opportunities at the same time?
SELECT Id, (SELECT id Opportunity) FROM Account
Doing a subquery